api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -501,10 +506,44 @@ public void validateNewProfile(Profile profile) throws BadRequestException {
     validateProfileForCorrectness(dummyProfile, profile);
   }
 
+  // Get all the profiles. Best we can do without surgery on the entity classes is to do one query
+  // per table and join them in code.
   public List<Profile> listAllProfiles() {
-    return userService.getAllUsers().stream().map(this::getProfile).collect(Collectors.toList());
+    final Set<DbUser> usersHeavy = userService.findAllUsersWithAuthoritiesAndPageVisits();
```

Comment:
nit: I'm not sure the `heavy` suffix helps clarify anything here. Is this important to a reader, and/or does it map to any well-defined concept or pattern in the codebase?

Reply:
We don't have an established convention, but I thought it important to indicate that this set of users had more fields populated than the default DbUser. The convention at HubSpot would be `hydratedUsers`.